x-pack/plugins/endpoint/server/plugin.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Plugin, CoreSetup, CoreStart } from 'kibana/server';
+import { addRoutes } from './routes';
+import { PluginSetupContract as FeaturesPluginSetupContract } from 'x-pack/plugins/features/server';
+
+export interface PluginsSetup {
+  features: FeaturesPluginSetupContract;
+}
+
+export class EndpointPlugin implements Plugin<void, void, PluginsSetup, {}> {
+  public setup(core: CoreSetup, plugins: PluginsSetup) {
+    plugins.features.registerFeature({
+      id: 'endpoint',
+      name: 'Endpoint',
+      icon: 'bug',
+      navLinkId: 'endpoint',
```

Comment:
@stacey-gammon @joshdover 
 Endpoint had no `registerFeature` call and as a result was showing up regardless of the user's role. I read some [docs](https://www.elastic.co/guide/en/kibana/current/development-plugin-feature-registration.html) and took a guess by adding this. The icon still shows up even when I login as a user that shouldn't have access to Endpoint.

I did a test by creating a role that should not have access to Endpoint:
![image](https://user-images.githubusercontent.com/35559/69360528-bca7f280-0c58-11ea-8f04-f743924740a0.png)
and then creating a user that has only that role:
![image](https://user-images.githubusercontent.com/35559/69360557-cb8ea500-0c58-11ea-8857-841d9ff27c15.png)

Logging in as this user shows the 'E' link in the nav for the Endpoint app:
![image](https://user-images.githubusercontent.com/35559/69360664-f4169f00-0c58-11ea-90e8-e5ae49097713.png)

The user cannot navigate to the app (as expected):
![image](https://user-images.githubusercontent.com/35559/69360679-fe389d80-0c58-11ea-9e01-800d16e678aa.png)

@joshdover Suggested that the link-hiding functionality of this system is not yet available in new-platform plugins. He recommended creating a legacy plugin to complement our new platform one.

@stacey-gammon @joshdover Could I have some help with this? Is this the right direction? How would I do it? Thanks!

Reply:
Yes this isn't implemented yet for New Platform applications, and I don't think there's a great way to workaround this right now. We're implementing this feature right now.

To be clear, when the plugin is disabled the icon is hidden, correct? It's just not hidden when the plugin is enabled and the user doesn't have access to Endpoint via their privileges?

Reply:
@joshdover Yes that's right. (but I'll confirm just to be sure)

Reply:
The plugin is disabled by default and the link is not there by default: 
![image](https://user-images.githubusercontent.com/35559/69379911-600bfe00-0c7f-11ea-86c9-beee1c72d4e7.png)
![image](https://user-images.githubusercontent.com/35559/69379944-69956600-0c7f-11ea-8698-7000a5386c04.png)
So I'd say that's working fine. Sorry for the confusion

Reply:
You could probably then just punt feature controls to a different PR.  There really is nothing about this app (esp as it is now "hello world") that would require feature controls in a first iteration.

Reply:
Yeah I agree. By the time we'll be putting this in front of users, the Platform changes will be in place to support feature controls here, without any changes required by the Endpoint app.